### PR TITLE
Backend preview

### PR DESCRIPTION
**Pull request by TimYuenior**
PreviewController improvements:
- Added Shot info box in the UI
- Fixed the highlight box with every way to navigate the shotbox and selecting shots
- Fixed an error with the PreviewController given when it is loaded with less than 4 shots loaded.
- Added error and gray images for invalid image paths and when there is no shot respectively.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java [hunk 1]
```diff
@@ -56,95 +58,106 @@
     @FXML private ImageView viewTwo;
     @FXML private ImageView viewThree;
     @FXML private ImageView viewFour;
+    
+    @FXML private Label camNum;
+    @FXML private Label presNum;
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java]: Naming could be better, presetNum for example.

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java [hunk 2]
```diff
@@ -56,95 +58,106 @@
...
     private Script script;
+    private List<Image> images;
 
     @FXML private void initialize() {
 
         script = ContextTFP.getScript();
         shots = script.getShots();
         
         currentShot = new SimpleIntegerProperty(1);
+        currentHighlight = 1;
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java]: When there are no shots, there shouldn't be a highlight right?

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java [hunk 3]
```diff
@@ -370,10 +400,11 @@ private int checkShots(Shot shot) {
     private void showShot(Shot shot) {
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java:399]: This method is still really ugly, but that wasn't the point of this branch. So I will ignore it for now.

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java [hunk 4]
```diff
@@ -383,32 +414,114 @@ private void switchViews(Shot shot) {
...
+        } else {
+            disableHighlight();
+        }
+    }
+    
+    /**
+     * Checks if the current view to be highlighted is in bounds.
+     */
+    private boolean highlightCheck() {
+        if (currentHighlight < 1 || currentHighlight > 4) {
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java:507]: Nice method, could be really useful!

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java [hunk 5]
```diff
@@ -56,95 +58,106 @@
...
         
         actualCam.fitWidthProperty().bind(imagePane.widthProperty());
         actualCam.fitHeightProperty().bind(imagePane.heightProperty());
-
+        
+        highlight1.setOpacity(0);
         highlight2.setOpacity(0);
         highlight3.setOpacity(0);
         highlight4.setOpacity(0);
     }
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java:104]: Can't you use your disableHighlight method for this?

Comment (TimYuenior) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java:104]: Nice catch, that can certainly be used here.

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java (hunk 2 above)
Comment (TimYuenior) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java]: True, will initialize to -1 to make it more clear.

File: src/main/java/nl/tudelft/contextproject/gui/PreviewController.java (hunk 1 above)
Comment (TimYuenior) [inline src/main/java/nl/tudelft/contextproject/gui/PreviewController.java]: Will change

**Comment by tkolenbrander** [comment]
Changes fixed my issues. When the build passes, I will merge the branch.

**Pull request merged**